src/server/cluster/cluster_slot_migration.cc

```diff
@@ -0,0 +1,64 @@
+// Copyright 2023, DragonflyDB authors.  All rights reserved.
+// See LICENSE for licensing terms.
+//
+#include "server/cluster/cluster_slot_migration.h"
+
+#include <absl/flags/flag.h>
+
+#include "base/logging.h"
+#include "server/error.h"
+#include "server/main_service.h"
+
+ABSL_FLAG(int, source_connect_timeout_ms, 20000,
+          "Timeout for establishing connection to a replication master");
+
+namespace dfly {
+
+using namespace std;
+using absl::GetFlag;
+
+ClusterSlotMigration::ClusterSlotMigration(string host, uint16_t port)
+    : ProtocolClient(std::move(host), port) {
+}
+
+ClusterSlotMigration::~ClusterSlotMigration() {
+}
+
+error_code ClusterSlotMigration::Start(ConnectionContext* cntx) {
+  VLOG(1) << "Starting slot migration";
+
+  auto check_connection_error = [this, &cntx](error_code ec, const char* msg) -> error_code {
+    if (ec) {
+      (*cntx)->SendError(absl::StrCat(msg, ec.message()));
+      cntx_.Cancel();
+    }
+    return ec;
+  };
+
+  VLOG(1) << "Resolving source host DNS";
+  error_code ec = ResolveMasterDns();
```

Comment:
you will get the ip in the start slot migration command, not the host name

Reply:
if I use localhost, it doesn't work without this call

Reply:
use 127.0.0.1 not localhost